source/cl/test/UnitCL/source/cl_khr_command_buffer/clCommandNDRangeKernelKHR.cpp

```diff
@@ -72,6 +72,55 @@ TEST_F(CommandNDRangeKernelTest, EmptyKernel) {
   ASSERT_SUCCESS(clReleaseProgram(program));
 }
 
+// Tests the whether we can enqueue an execute a command buffer containing a
+// kernel enqueue created via clCommandNDRangeKernelKHR with global size of 0
+// or a global size of nullptr
+TEST_F(CommandNDRangeKernelTest, GlobalZero) {
+  // Requires a compiler to compile the kernel.
+  if (!getDeviceCompilerAvailable()) {
+    GTEST_SKIP();
+  }
+  // Set up the kernel.
+  const char *code = R"OpenCLC(
+  __kernel void empty_kernel() { __builtin_trap(); }
+)OpenCLC";
+  const size_t code_length = std::strlen(code);
+
+  cl_int error = CL_SUCCESS;
+  cl_program program =
+      clCreateProgramWithSource(context, 1, &code, &code_length, &error);
+  ASSERT_SUCCESS(error);
+  ASSERT_SUCCESS(
+      clBuildProgram(program, 1, &device, nullptr, nullptr, nullptr));
+
+  cl_kernel kernel = clCreateKernel(program, "empty_kernel", &error);
+  ASSERT_SUCCESS(error);
+
+  // Set up the command buffer and run the command buffer.
+  cl_command_buffer_khr command_buffer =
+      clCreateCommandBufferKHR(1, &command_queue, nullptr, &error);
+  ASSERT_SUCCESS(error);
+
+  constexpr size_t global_size = 0;
+  ASSERT_SUCCESS(clCommandNDRangeKernelKHR(
+      command_buffer, nullptr, nullptr, kernel, 1, nullptr, &global_size,
+      nullptr, 0, nullptr, nullptr, nullptr));
+
+  ASSERT_SUCCESS(clCommandNDRangeKernelKHR(
+      command_buffer, nullptr, nullptr, kernel, 1, nullptr,
+      /* global size */ nullptr, nullptr, 0, nullptr, nullptr, nullptr));
```

Comment:
Let's use the LLVM convention here that I personally would prefer us to use more consistently:
```suggestion
      /*global_work_size=*/nullptr, nullptr, 0, nullptr, nullptr, nullptr));
```